### DISCUSSION TITLE
Great :-)

### DISCUSSION THREAD

**Discussion by iacchos-fr** [JuergenLeber/SHIFTR#9] [General]
Hi @JuergenLeber 

Very good job, I just finished the installation of my SHIFTR for a Tacx Neo2T, it woks fine !!
I am not at all a expert in ESP32 programming, but I succeed to build my SHIFTR box with (almost) any issue :-D

I have some comments / questions : 
- last release is 1.2, but 1.1.1 version is displayed on local web server page (see version.h file)
- you specified to use 3.3v on USB/TTL converter for installation : it also works with 5V (I do it) ?
- for future firmware update, the file to upload is the one provided on the release page (SHITR-v1.2.0.bin for last release) ?
- I haven't an Ethernet connection near my smart trainer, only a Wifi connection. I managed to connect Zwift to the SHIFTR box through the AP wifi server (SHIFTR.12345). But maybe it could be a smart feature to provide a connection of the WT32-ETH01 to the local Wifi network ?

Thank again for the job
Regards

**Comment by JuergenLeber** [comment]
Hi @iacchos-fr,

great to hear that you got it to work, congratulations! I hope you meant "without any issues" and not "with" ;-)

Regarding your questions:

> last release is 1.2, but 1.1.1 version is displayed on local web server page (see version.h file)
- That's because the release is built via GitHub Actions on their servers and the version.h file get's created there and not committed back. Installation via PlatformIO is only necessary for the first run. All future upgrades can be done via web interface.

> you specified to use 3.3v on USB/TTL converter for installation : it also works with 5V (I do it) ?
- Sure, if you connect to the 5V pins on the WT32-ETH01 it will also work but just do an additional conversion from 5 to 3.3V. But the efficiency loss is so small that you can also go with 5V.

> for future firmware update, the file to upload is the one provided on the release page (SHITR-v1.2.0.bin for last release) ?
- Exactly - This one will then also show the correct version information on the status page

> I haven't an Ethernet connection near my smart trainer, only a Wifi connection. I managed to connect Zwift to the SHIFTR box through the AP wifi server (SHIFTR.12345). But maybe it could be a smart feature to provide a connection of the WT32-ETH01 to the local Wifi network ?
- That should work - although it's untested. You can just connect via AP first and then go to http://shiftr-12345.local/config. This will look a bit strange but it is the config interface from the wifi manager I used. There you can set WIFI SSID and WIFI Password and it will connect to it. Please be aware that the ESP32 only supports 2.4GHz wifi. After that the AP mode will not be available anymore but the .local hostname should stay in your network. Attention: If you set a new AP password there this one will then also be necessary for the web interface (user admin/pass AP password). Please report if it worked! 

Regards,
Jürgen

**Comment by iacchos-fr** [comment]
Thanks Jürgen for reply !
I tried with  http://shiftr-12345.local/config and was able to connect the ESP32 to my Wifi network ! I am able to connect to the .local hostname from my Zwift PC... but Zwift is no more able to find the SHIFTR ? I do not understand why... I will investigate further this week !
Regards

**Reply by iacchos-fr** [reply]
I tried again this morning: it works great :-) Much easier to ensure a permanent wireless connection between my PC and the SHIFTR!
I think it's worth documenting this feature!

Thank again for that insane job !

**Comment by iacchos-fr** [comment]
Hi @JuergenLeber

2 or 3 points I wanted to share :-)

- I ordered a WT32-ETH01 module from a well-known Chinese site... and I received an ESP32-ETH01 (the ‘economy’ version...). It's very compatible with the WT32 and SHITR works very well on it... but this model doesn't fit in the 3D print box... because it's 3mm longer :-( I had to buy a ‘real’ module and redo the installation...
- I tried to program the module at 3.3V, but couldn't (no USB/TTL Uart connection and the module?). I succeeded with 5V
- I drilled a small hole in the 3D print box... to see the module's power on LED :-)

![IMG_1288](https://github.com/user-attachments/assets/59d3a037-9068-4716-b388-d56a4e3fa962)

**Reply by JuergenLeber** [reply]
Could you please share a link to the "3mm longer" module? I have only seen the normal "WT32-ETH01" and a new one "ETH01-EVO" so far but the "evo"-one uses a totally different ethernet chip.

**Reply by iacchos-fr** [reply]
https://fr.aliexpress.com/item/1005006583793993.html

The chip is referenced as "ESP32-VROOM-32"

![S1c9fc1d49ebb48388cef524a9eaaa604K jpg](https://github.com/user-attachments/assets/bdde1442-8e00-4540-91ca-611ce866da12)

**Reply by JuergenLeber** [reply]
Strange - 6.3cm should fit perfectly - that's the same size as I have.

**Reply by iacchos-fr** [reply]
Yes strange... The WT32-ETH01 is 60mm only
![mod-wt32-eth01_dimensions_zl](https://github.com/user-attachments/assets/5ed3edd5-8571-4942-a2e8-4156e0980ac1)

**Comment by JuergenLeber** [comment]
Just measured mine (have some of them...) and it's exactly as in your image (even if the page on Amazon stated other dimensions)

So @ everybody: DON'T buy "ESP32-VROOM-32", only the original WT32-ETH01.

**Comment by JuergenLeber** [comment]
Closing this one as resolved.
